db/migrate/20240624185108_reactivate_users_and_remove_org_user_roles.rb

```diff
@@ -0,0 +1,15 @@
+class ReactivateUsersAndRemoveOrgUserRoles < ActiveRecord::Migration[7.1]
+  def up
+    users = User.unscoped.joins(:roles).where.not(discarded_at: nil).where("roles.name": "org_user")
+    users.each do |user|
+      user.transaction do
+        user.update!(discarded_at: nil)
+        user.roles.where(name: "org_user").delete_all
+      end
+    end
+  end
```

Comment:
I checked the data, and every discarded user has an org\_user role. This makes sense, because you can only discard a user from the organization user management page. So I think you're safe to not have to check the role name.

In fact, you should just delete _all_ roles that belong to current discarded users - many of them also have partner roles, but if they're discarded, they wouldn't have been able to log in. So effectively all their roles should be removed.

Reply:
Updated this. Since all discarded users are org users might as well simplify it further and just deal exclusively with discarded users.